fastlane/Fastfile

```diff
@@ -38,80 +68,182 @@ platform :ios do
+      build_configuration: build_configurations[:testflight],
+      build_setting: 'CURRENT_PROJECT_VERSION',
+      build_setting_value: next_build_number
+    )
 
-    # We increment the build number. If you are adding fastlane to an existing project, check what the current build is in itunes connect. 
-    # For example, if it is 15, you should set in xcode build number 15 (because increment_build_number increments the build).
-    increment_build_number
+    # Back up the build number in Info.plist.
+    info_plist_backup_build_number = get_info_plist_value(
+      path: project_plist_path % ENV['PROJECT_NAME'],
+      key: 'CFBundleVersion'
+    )
+
+    # Update the build number in Info.plist.
+    set_info_plist_value(
+      path: project_plist_path % ENV['PROJECT_NAME'],
+      key: 'CFBundleVersion',
+      value: next_build_number
+    )
+
+    # Bundle identifier from xCode project
+    bundle_identifier = read_project_property(
+      build_configuration: build_configurations[:testflight],
+      build_setting: 'PRODUCT_BUNDLE_IDENTIFIER'
+    )
 
     # Build
     build_app(
-      # Set the app id
-      app_identifier: ENV["APP_IDENTIFIER"],
-      # What configuration to use, usefull for keeping different API keys etc between environments
-      configuration:"Release",
-      # Use this codesigning identity (this is the name of the certificate in your keychain)
-      codesigning_identity: ENV["CODE_SIGNING_ENTITY"],
-      # the projectname, this is the name of the .xcodeproj file and the folder containing your code in the project
-      project_name: ENV["PROJECT_NAME"],
-      # the scheme to build
-      scheme: "",
-      # Your team ID
-      team_id: ENV["TEAM_ID"]
-    )
-
-    upload_dsym_to_rollbar(
-      rollbar_url: 'https://api.rollbar.com/api/1/dsym',
-      access_token: ENV["ROLLBAR_POST_SERVER_ITEM_ACCESS_TOKEN"],
-      version: get_info_plist_value(path: '', key: 'CFBundleVersion'), # complete with info.plist path. ex: 'Project/Info.plist'
-      bundle_identifier: ENV["APP_IDENTIFIER"],
-      dsym: '' # DSYM location with @. Example: '@../wolox.app.dSYM.zip'. @ means that it takes the content of the file.
+      app_identifier: bundle_identifier,
+      configuration: build_configurations[:testflight],
+      match_type: match_types[:testflight],
+      project_name: ENV["PROJECT_NAME"]
+    )
+
+    # Get rollbar server access token from configuration file.
+    rollbar_server_access_token = read_xcconfig_property(
+      xcconfig_path: project_xcconfig_path % [ENV['PROJECT_NAME'], build_configurations[:testflight]],
+      xcconfig_key: 'ROLLBAR_SERVER_ACCESS_TOKEN'
+    )
+
+    # Upload dsym to rollbar.
+    upload_dsym(
+      access_token: rollbar_server_access_token,
+      version: next_build_number,
+      bundle_identifier: bundle_identifier,
+      dsym_zip_path: dsym_zip_path % ENV['PROJECT_NAME']
+    )
+
+    # Puts back the backed up build number in Info.plist.
+    set_info_plist_value(
```

Comment:
Because each configuration uses a different bundle version. Each of them are stored in xcode build settings under the key 'CURRENT_PROJECT_VERSION'.

The `Info.plist` is a single file shared and used by every configuration, so I didn't want it to have a version related to a particular build configuration. Instead this `Info.plist` is changed when building the app, and then these changes are discarded.